### ISSUE TITLE
Backpropagation skew iou

### ISSUE THREAD

**Issue by kelvinkoh0308** [ming71/rotate-yolov3#3]
Hi, does backprop works on you skew iou calculation? This is because from i saw that you were using shapely in your calculation of skewiou which is not tensors

**Comment by ming71** [comment]
Sorry, I forget it, it has been a long term last time I ran this code. And it seems no backprop  conducted here.
I'll release a codebase with cuda riou calculation later. You can also modify yourself( is it the reason for not performing excellently here?maybe ).

**Comment by kelvinkoh0308** [comment]
@ming71 Thanks your reply, I am looking for the implementation of skewiou in either numpy,tf or pytorch but most of the source code online are using shapely package

**Comment by ming71** [comment]
> @ming71 Thanks your reply, I am looking for the implementation of skewiou in either numpy,tf or pytorch but most of the source code online are using shapely package

Hello, I am also thinking about this issue recently. In fact, there is no need to obtain the gradient if skewiou is only used as a standard for label assignment . But if it is integrated into loss function, then it needs to be derived. There are two solutions:
1. Derive the back-propagation formula and implement it with CUDA
2. Use built-in method of pytorch to achieve riou calculation (autograd helps).
What is your opinion? Have you solved this problem?

**Comment by kelvinkoh0308** [comment]
Hi
I haven't solve the problem because I am not familiar to implement it with cuda, I also did try to search the riou calculation in python but none of the implementation is in python.
